kuma/static/js/contribution-handler.js

```diff
@@ -0,0 +1,513 @@
+            }
+
+            return;
+        }
+
+        // Send GA Event.
+        mdn.analytics.trackEvent({
+            category: 'Contribution submission',
+            action: isPopoverBanner ? 'On Popover' : 'On Page',
+            value: 1
+        });
+
+        // On success open Stripe Checkout modal.
+        stripeHandler.open({
+            image: 'https://avatars1.githubusercontent.com/u/7565578?s=280&v=4',
+            name: 'MDN Web Docs',
+            description: 'Contribute to MDN Web Docs',
+            zipCode: true,
+            amount: (selectedAmount * 100),
+            email: $(emailField).val(),
+            closed: function() {
+                form.removeClass('disabled');
+            }
+        });
+    }
+
+    /**
+     * Handles the form button click. This will either attempt to submit the form
+     * or will expand the popover depending on the current state of the popover.
+     */
+    function onFormButtonClick() {
+        // Calculate the role of the submit button
+        if (isPopoverBanner && popoverBanner.hasClass('is-collapsed')) {
+            expandCta();
+        } else {
+            onSubmit();
+        }
+    }
+
+    var mediaQueryList = null;
+
+    /**
+     * Increases or decreases the height of the `popoverBanner`
+     * based on a `mediaQueryList` match
+     */
+    function handleViewportChange(evt) {
+        if (evt.matches) {
+            popoverBanner.removeClass('expanded');
+            popoverBanner.addClass('expanded-extend');
+        } else {
+            popoverBanner.removeClass('expanded-extend');
+            popoverBanner.addClass('expanded');
+        }
+    }
+
+    /**
+     * Expands the popover to show the full contents.
+     */
+    function expandCta() {
+        var secondaryHeader = popoverBanner[0].querySelector('h4');
+        var smallDesktop = '(max-width: 1092px)';
+
+        mediaQueryList = window.matchMedia(smallDesktop);
+        var initialExpandedClass = mediaQueryList.matches ? 'expanded-extend' : 'expanded';
+
+        popoverBanner.addClass(initialExpandedClass + ' is-expanding');
+        popoverBanner.removeClass('is-collapsed');
+
+        // listens for, and responds to, matchMedia events
+        mediaQueryList.addListener(handleViewportChange);
+
+        popoverBanner.on('transitionend', function() {
+            popoverBanner.removeClass('is-expanding');
+            popoverBanner.attr('aria-expanded', true);
+
+            // store the current activeElement
+            activeElement = document.activeElement;
+            // move focus to the secondary header text
+            secondaryHeader.focus();
+
+            // remove the event listener
+            popoverBanner.off('transitionend');
+
+            // listen to minimise button clicks.
+            collapseButton.click(collapseCta);
+
+            $(doc).on('keydown.popoverCloseHandler', function(event) {
+                if (event.keyCode === 27) { // Escape key.
```

Comment:
I'd rather use ``event.key == 'Escape'``, but I think we want to support some unsupported browsers:

* https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key#Browser_compatibility
* https://caniuse.com/#feat=keyboardevent-key
* https://stackoverflow.com/a/3369743/10612